src/conn.c

```diff
@@ -2134,29 +2142,30 @@ _evStopPolling(natsConnection *nc)
     return s;
 }
 
-// _processOpError handles errors from reading or parsing the protocol.
+// _tryReconnect handles errors from reading or parsing the protocol, or forced
+// reconnection. It will fire off a doReconnect thread if needed.
 // The lock should not be held entering this function.
-static bool
-_processOpError(natsConnection *nc, natsStatus s, bool initialConnect)
+static natsStatus
+_tryReconnect(natsConnection *nc, natsStatus newErr, bool forcedReconnect, bool *started)
 {
+    natsStatus s = NATS_OK;
+
     natsConn_Lock(nc);
 
-    if (!initialConnect)
+    if (!forcedReconnect)
     {
         if (_isConnecting(nc) || natsConn_isClosed(nc) || (nc->inReconnect > 0))
         {
             natsConn_Unlock(nc);
 
-            return false;
+            return NATS_OK;
         }
     }
 
     // Do reconnect only if allowed and we were actually connected
     // or if we are retrying on initial failed connect.
-    if (initialConnect || (nc->opts->allowReconnect && (nc->status == NATS_CONN_STATUS_CONNECTED)))
+    if (forcedReconnect || (nc->opts->allowReconnect && (nc->status == NATS_CONN_STATUS_CONNECTED)))
     {
-        natsStatus ls = NATS_OK;
```

Comment:
There was a reason we used a different status here. What we wanted to report as the error was the error that was given to processOpError(), not an internal one that we may encounter trying to setup the reconnect.

Reply:
right, and I think my code still does that (`newErr`), but in the case of a natsConnection_Reconnect failing on a to fire off doReconnect we want to return the actual failure, right? (like, out of memory for instance)